monai/data/box_utils.py

```diff
@@ -0,0 +1,565 @@
+                - "xxyy": boxes has format [xmin, xmax, ymin, ymax]
+                - "xxyyzz": boxes has format [xmin, xmax, ymin, ymax, zmin, zmax]
+                - "xyxyzz": boxes has format [xmin, ymin, xmax, ymax, zmin, zmax]
+                - "xywh": boxes has format [xmin, ymin, xsize, ysize]
+                - "xyzwhd": boxes has format [xmin, ymin, zmin, xsize, ysize, zsize]
+                - "ccwh": boxes has format [xcenter, ycenter, xsize, ysize]
+                - "cccwhd": boxes has format [xcenter, ycenter, zcenter, xsize, ysize, zsize]
+            #. BoxMode class: choose from the subclasses of :class:`~monai.data.box_utils.BoxMode`, for example,
+                - CornerCornerModeTypeA: equivalent to "xyxy" or "xyzxyz"
+                - CornerCornerModeTypeB: equivalent to "xxyy" or "xxyyzz"
+                - CornerCornerModeTypeC: equivalent to "xyxy" or "xyxyzz"
+                - CornerSizeMode: equivalent to "xywh" or "xyzwhd"
+                - CenterSizeMode: equivalent to "ccwh" or "cccwhd"
+            #. BoxMode object: choose from the subclasses of :class:`~monai.data.box_utils.BoxMode`, for example,
+                - CornerCornerModeTypeA(): equivalent to "xyxy" or "xyzxyz"
+                - CornerCornerModeTypeB(): equivalent to "xxyy" or "xxyyzz"
+                - CornerCornerModeTypeC(): equivalent to "xyxy" or "xyxyzz"
+                - CornerSizeMode(): equivalent to "xywh" or "xyzwhd"
+                - CenterSizeMode(): equivalent to "ccwh" or "cccwhd"
+            #. None: will assume mode is ``StandardMode``
+
+    Returns:
+        bounding boxes with target mode, with same format as ``boxes``, does not share memory with ``boxes``
+
+    Example:
+        .. code-block:: python
+
+            boxes = torch.ones(10,4)
+            # The following three lines are equivalent
+            # They convert boxes with format [xmin, ymin, xmax, ymax] to [xcenter, ycenter, xsize, ysize].
+            box_convert_mode(boxes=boxes, src_mode="xyxy", dst_mode="ccwh")
+            box_convert_mode(boxes=boxes, src_mode="xyxy", dst_mode=monai.data.box_utils.CenterSizeMode)
+            box_convert_mode(boxes=boxes, src_mode="xyxy", dst_mode=monai.data.box_utils.CenterSizeMode())
+    """
+    src_boxmode = get_boxmode(src_mode)
+    dst_boxmode = get_boxmode(dst_mode)
+
+    # if mode not changed, deepcopy the original boxes
+    if isinstance(src_boxmode, type(dst_boxmode)):
+        return deepcopy(boxes)
+
+    # convert box mode
+    # convert numpy to tensor if needed
+    boxes_t, *_ = convert_data_type(boxes, torch.Tensor)
+
+    # convert boxes to corners
+    corners = src_boxmode.boxes_to_corners(boxes_t)
+
+    # check validity of corners
+    spatial_dims = get_spatial_dims(boxes=boxes_t)
+    for axis in range(0, spatial_dims):
+        if (corners[spatial_dims + axis] < corners[axis]).sum() > 0:
+            raise ValueError("Given boxes has invalid values. The box size must be non-negative.")
```

Comment:
```suggestion
            warnings.warn("Given boxes have invalid values. The box size must be non-negative.")
```
perhaps this could be a warning for now? maybe it'll be useful in some applications?